src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -233,4 +234,28 @@ private static int identifier(String s, int startFrom) {
     localVars.remove(loop.getVariable());
     return ret;
   }
+
+  private static OutputView.AssignmentView assignmentView(
+      String assign,
+      MethodContext context,
+      SampleValueSet valueSet,
+      Map<String, TypeModel> localVars) {
+
+    int p = identifier(assign, 0);
+    String ident = assign.substring(0, p);
+    Preconditions.checkArgument(
+        assign.startsWith("=", p),
+        "%s:%s invalid assign, expecting '=': %s",
+        context.getMethodModel().getSimpleName(),
+        valueSet.getId(),
+        assign);
+    OutputView.VariableView reference =
+        accessor(assign.substring(p + 1), context, valueSet, localVars, ident);
+    return OutputView.AssignmentView.newBuilder()
```

Comment:
Actually, my last comment on this thread is not quite correct if you implement a local variable table. To put the restriction in place coherently, you would need to check that a new variable wan't defined even inside a block that is not in scope now.

---

Another thought: another way to implement block scope is, instead of cloning the variable table, having something akin to a stack of variable tables corresponding to the scopes that are on the stack. I say "akin" because you'd want to easily search for a given variable name at all levels of the stack. (If we were doing proper block scoping allowing duplicate identifiers in different scopes, you would do this search from the top of the stack down.)